src/main/java/com/gluonhq/substrate/target/IosTargetConfiguration.java

```diff
@@ -41,33 +41,80 @@
 import java.io.IOException;
 import java.nio.file.Files;
 import java.nio.file.Path;
+import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.Collections;
 import java.util.List;
 
 public class IosTargetConfiguration extends AbstractTargetConfiguration {
 
-    private List<String> iosAdditionalSourceFiles = Collections.singletonList("AppDelegate.m");
+    private List<String> iosAdditionalSourceFiles = Arrays.asList("AppDelegate.m", "thread.m");
+
+    private static final List<String> javafxReflectionIosClassList = Arrays.asList(
+            "com.sun.prism.es2.ES2Pipeline",
+            "com.sun.prism.es2.IOSGLFactory",
+            "com.sun.javafx.font.coretext.CTFactory",
+            "com.sun.scenario.effect.impl.es2.ES2ShaderSource",
+            "com.sun.glass.ui.ios.IosApplication",
+            "com.sun.glass.ui.ios.IosCursor",
+            "com.sun.glass.ui.ios.IosGestureSupport",
+            "com.sun.glass.ui.ios.IosMenuBarDelegate",
+            "com.sun.glass.ui.ios.IosPlatformFactory",
+            "com.sun.glass.ui.ios.IosPixels",
+            "com.sun.glass.ui.ios.IosView",
+            "com.sun.glass.ui.ios.IosWindow"
+    );
+
+    private static final List<String>javafxJNIIosClassList = Arrays.asList(
+            "com.sun.glass.ui.ios.IosApplication",
+            "com.sun.glass.ui.ios.IosCursor",
+            "com.sun.glass.ui.ios.IosGestureSupport",
+            "com.sun.glass.ui.ios.IosMenuBarDelegate",
+            "com.sun.glass.ui.ios.IosPixels",
+            "com.sun.glass.ui.ios.IosView",
+            "com.sun.glass.ui.ios.IosWindow",
+            "java.util.Vector",
+            "com.sun.javafx.font.coretext.CGAffineTransform",
+            "com.sun.javafx.font.coretext.CGPoint",
+            "com.sun.javafx.font.coretext.CGRect",
+            "com.sun.javafx.font.coretext.CGSize",
+            "com.sun.javafx.font.FontConfigManager$FcCompFont",
+            "com.sun.javafx.font.FontConfigManager$FontConfigFont",
+            "com.sun.javafx.iio.ios.IosImageLoader"
+    );
+
+    private static final List<String> ioslibs = Arrays.asList(
+            "-lpthread", "-lz", "-lstrictmath", "-llibchelper",
+            "-ljava", "-ljvm", "-lnio", "-lzip", "-lnet");
+
+    private static final List<String> javafxLibs = Arrays.asList(
+            "prism_es2", "glass", "javafx_font", "prism_common", "javafx_iio");
 
     @Override
     List<String> getTargetSpecificLinkFlags(boolean useJavaFX, boolean usePrismSW) {
-        return Arrays.asList("-w", "-fPIC",
+        List<String> linkFlags = new ArrayList<>(Arrays.asList("-w", "-fPIC",
```

Comment:
Would just `Arrays.asList` be enough? Why do we need list in the list?

Reply:
Because `Arrays.asList` returns a list implementation that doesn't support `addAll`, and therefore `linkFlags.addAll(ioslibs);` will fail.

Reply:
I missed that :)